pkgs/servers/nosql/mongodb/6.0.nix

```diff
@@ -9,6 +9,7 @@ buildMongoDB {
   version = "6.0.13";
   sha256 = "sha256-BD3XrTdv4sCa3h37o1A2s3/R0R8zHiR59a4pY0RxLGU=";
   patches = [
+    ./fix-gcc-13-ctype-6_0.patch
```

Comment:
Each patch in nixpkgs applied to the upstream source should be documented. Out-of-tree patches, fetched using `fetchpatch`, are preferred.

In each patch comment, please explain the purpose of the patch and link to the relevant upstream issue if possible. If the patch has been merged upstream but is not yet part of the released version, please note the version number or date in the comment such that a future maintainer updating the nix expression will know whether the patch has been incorporated upstream and can thus be removed from nixpkgs.

Furthermore, please use a _stable_ URL for the patch. Rather than, for example, linking to a GitHub pull request of the form `https://github.com/owner/repo/pull/pr_number.patch`, which would change every time a commit is added or the PR is force-pushed, link to a specific commit patch in the form `https://github.com/owner/repo/commit/sha.patch`.

Here are two good examples of patch comments:

```nix
mkDerivation {
  …

  patches = [
    # Ensure RStudio compiles against R 4.0.0.
    # Should be removed next 1.2.X RStudio update or possibly 1.3.X.
    (fetchpatch {
      url = "https://github.com/rstudio/rstudio/commit/3fb2397c2f208bb8ace0bbaf269481ccb96b5b20.patch";
      sha256 = "0qpgjy6aash0fc0xbns42cwpj3nsw49nkbzwyq8az01xwg81g0f3";
    })
  ];
}
```

```nix
mkDerivation {
  …

  patches = [
    # Allow building with bison 3.7
    # PR at https://github.com/GoldenCheetah/GoldenCheetah/pull/3590
    (fetchpatch {
      url = "https://github.com/GoldenCheetah/GoldenCheetah/commit/e1f42f8b3340eb4695ad73be764332e75b7bce90.patch";
      sha256 = "1h0y9vfji5jngqcpzxna5nnawxs77i1lrj44w8a72j0ah0sznivb";
    })
  ];
}
```

Reply:
Have you had a look on any of the other patches in MongoDB?

Reply:
Are you trying to tell me that the reason why you don't want to do it is because it was not done correctly somewhere else?

Reply:
You can get an comment in the .nix that explains what the patch is doing. 
But I really dont get the point using an additional fetchpatch with an url to github (where then always an connection to github must be opened) instead of including it locally in the repo like all the other patched.
Reminder that mongo is not free software and has to be build every time instead of downloading it from cache.nixos.org.

This patch affects just the 6.0 version (since this commit https://github.com/NixOS/nixpkgs/commit/f29d66a77f6f5de17158037eef7e52b4f555ce77), hence it is included in the 6.0.nix file.

Reply:
I mean, is there an upstream commit we can fetch or not?

Reply:
No there isn't. I built the patch based on an error message / suggestion that got thown while building the software.

Reply:
Then we can't fetchpatch something.

Reply:
right. Thats why it is just the file in the repo next to the 6.0.nix